src/connect4/core.rs

```diff
@@ -357,4 +417,243 @@ pub fn main() -> GameResult {
 
     let state = &mut GameState::new(ctx)?;
     event::run(ctx, events_loop, state)
-}
\ No newline at end of file
+}
+
+
+#[cfg(test)]
+mod core_tests {
+    use super::*;
+    mod Board {
+        use super::*;
+        use connect4::core::Board;
+
+        //Method to create a board state from a set of vectors, where 0 is empty and 1 or 2 team tokens
+        //Note that input is board[column][row], so if you want to add a team 1 token in column 4, row 0, then
+        //the board input should have board[4][0] = 1
+        fn create_test_board(board: Vec<Vec<i32>>) -> Board {
+            let mut output = Board::new(GridPosition{ x: 0, y:0 });
+            for i in 0..BOARD_SIZE.1 {
+                if (i as usize) < board.len() {
+                    let col = board.get(i as usize).unwrap();
+                    for j in 0..BOARD_SIZE.0 {
+                        if (j as usize) < col.len() {
+                            let val = *col.get(j as usize).unwrap();
+                            if val > 0 {
+                                output.insert(i, val, MyColor::White);
```

Comment:
This is an example of why it'd be like to have a `Team` struct with a team number and color - had to pass in a random `MyColor::White` to `insert` when all I care about is which team it is.